assets/model_monitoring/components/src/model_data_collector_preprocessor/store_url.py

```diff
@@ -190,10 +193,16 @@ def _set_properties(self, ws: Workspace):
             self.path = matches.group("path").strip("/")
             self._datastore = None
         elif url.scheme == "azureml":
-            if ':' in url.path:  # azureml asset path
+            if ':' in url.path:  # azureml:<data_name>:<version> asset path
                 # asset path should be translated to azureml or hdfs path in service, should not reach here
                 raise InvalidInputError("AzureML asset path is not supported as uri_folder.")
-            else:  # azureml long or short form
+
+            data_pattern0 = r"azureml://(subscriptions/([^/]+)/resourceGroups/([^/]+)/workspaces/([^/]+)/)?data/([^/]+)/versions/(.+)"  # noqa: E501
+            data_pattern1 = r"azureml://locations/([^/]+)/workspaces/([^/]+)/data/([^/]+)/versions/(.+)"
+            if re.match(data_pattern0, self._base_url) or re.match(data_pattern1, self._base_url):
+                raise InvalidInputError("AzureML data asset url is NOT supported as input of Model Monitoring job, "
+                                        "please use credential datastore + path instead.")
```

Comment:
Is there some more clear instruction we can give? Maybe just tell them to use short form (`azureml:<name>:<version>`) or long form since we know that should work. I think as a user if I got this error on my MoMo job I might not know how to "... use credential datastore + path ..." as input